server/server.go

```diff
@@ -56,36 +56,23 @@ type ServerEventListener interface {
 
 // NewServer creates a server with the given protocol, address, authentication
 // details given a SQLe engine and a session builder.
-func NewServer(cfg Config, e *sqle.Engine, sb SessionBuilder, listener ServerEventListener) (*Server, error) {
-	var tracer trace.Tracer
-	if cfg.Tracer != nil {
-		tracer = cfg.Tracer
-	} else {
-		tracer = sql.NoopTracer
-	}
-
-	sm := NewSessionManager(sb, tracer, e.Analyzer.Catalog.Database, e.MemoryManager, e.ProcessList, cfg.Address)
-	handler := &Handler{
-		e:                 e,
-		sm:                sm,
-		readTimeout:       cfg.ConnReadTimeout,
-		disableMultiStmts: cfg.DisableClientMultiStatements,
-		maxLoggedQueryLen: cfg.MaxLoggedQueryLen,
-		encodeLoggedQuery: cfg.EncodeLoggedQuery,
-		sel:               listener,
-	}
-	//handler = NewHandler_(e, sm, cfg.ConnReadTimeout, cfg.DisableClientMultiStatements, cfg.MaxLoggedQueryLen, cfg.EncodeLoggedQuery, listener)
-	return newServerFromHandler(cfg, e, sm, handler, listener)
+func NewServer(cfg Config, e *sqle.Engine, ctxFactory sql.ContextFactory, sb SessionBuilder, listener ServerEventListener) (*Server, error) {
```

Comment:
I don't love having both a ctxFactory and a SessionBuilder here, the reality is these two responsibilities are closely related if not overlapping. Ideally this would be a single real interface with two methods that a single type could implement. This isn't a deal breaker but as long as you're monkeying around in here it might be worthwhile to see what it would take to get there.

Reply:
Very much agree. This ContextFactory is a pretty weird integration point as of now. There are other ways of achieving what we need. For example, we could allow an integrator to pass along a `[]sql.ContextOption` instead, and make a new context option like `WithContextTransform(func(context.Context) context.Context)`. The reality is that the lifecycle around all these things is really subtle right now, and making changes is somewhat fragile and far reaching.

This change as is is useful to me and I have doltgres and dolt fixes ready for it, so I'm going to go ahead and land it, but I may revisit it soon.